src/plugins/plugins_ex.h

```diff
@@ -392,10 +392,20 @@ class pluginex : public plugin
 
     virtual ~pluginex()
     {
-        for (auto& trap : traps)
-            delete_trap(trap);
+        stop();
     };
 
+    virtual void stop()
+    {
+        while (!traps.empty())
+            destroy_trap(traps.front());
+    }
+
+    virtual bool is_stopped()
+    {
+        return true;
```

Comment:
Same here, I think the state of the plugin needs to be tracked and not just return true by default.

Reply:
Here we have to provide some default implementation. So I tried to not break current logic.

Reply:
Thank You! I have fixed logical error.